contrib/go/src/python/pants/contrib/go/tasks/go_fetch.py

```diff
@@ -47,7 +53,27 @@ def execute(self):
     if not go_remote_libs:
       return
 
-    undeclared_deps = self._transitive_download_remote_libs(set(go_remote_libs))
+    # We cache mappings from import path to root (e.g., example.org/pkg/foo -> example.org),
+    # otherwise we may have to fetch them over the network via the meta tag protocol.
+    # Note that this caching is unversioned: The mapping is defined as "whatever meta tag is
+    # currently being served at the relevant URL", which is inherently unversioned.
+    import_root_map_path = os.path.join(self.workdir, 'pkg_root_map.txt')
```

Comment:
This file won't be cached, but I think that the per-target information will be, due to being located under the `target.results_dir`.

Would it be worthwhile to cache this on a per-target basis (ie, redundantly) so that it can be fetched from cache?